js/src/dom/event-handler.js

```diff
@@ -171,6 +164,24 @@ function addHandler(element, originalTypeEvent, handler, delegationFn, oneOff) {
     delegationFn = null
   }
 
+  // in case of mouseenter or mouseleave wrap the handler within a function that checks for its DOM position
+  // this prevents the handler getting triggered the same way as mouseover or mouseout does
+  if (/^mouse(enter|leave)/i.test(originalTypeEvent)) {
```

Comment:
hmm, maybe you can choose something more simple, like  (mouseenter|mouseleave) to help on future searches